src/core/instance/events.js

```diff
@@ -21,6 +22,8 @@ export function initEvents (vm: Component) {
 
 let target: Component
 
+const updateEventRE = /^update:([\w.$]+)\.([\w$]+)$/
```

Comment:
Should we do `^update:([\w.$]+[^.])\.([\w$]+)$` to prevent things like `update:foo..bar`